File: HitandBlow_C/HitandBlow.c
#include<stdio.h>
#include<stdlib.h>
#include<time.h>
#include<string.h>
#include<ctype.h>

#define NUM 3

void randNumber(int rand_ans[]);
int numCheck(const char user_ans[]);
int readUserAns(const char user_ans[], const int com_ans[], int user_eat, int user_bite);

int main(){

    int cnt = 1;
    int end = 1;
    int eat, bite, check;
    int com_ans[NUM];
    char buff[10];

    printf("***Hit & Blow***\n");
    randNumber(com_ans);

    while(end != 0){

        printf("%d回目の挑戦\n", cnt);
        printf("%dつの数字を入力してください >>>", NUM);
        scanf("%s", buff);
        check = numCheck(buff);

            switch (check){
                case 1:
                    printf("%dつの数字を入力してください\n", NUM);
                    break;
                case 2:
                    printf("数字以外の文字が含まれてます\n");
                    break;
                case 3:
                    printf("数字が重複しています\n");
                    break;
                default:
                    break;
        }
        int eat = readUserAns(buff, com_ans, eat, bite);

        if(eat == NUM){
            printf("Congratulations!! ゲームを終了します\n");
            end = 0;
        }
        cnt++;
    }
    return 0;
}

void randNumber(int rand_ans[]){

    int i, j, rand_num;
    srand((unsigned)time(NULL));

    for (i = 0; i < NUM; i++) {
        do {
            rand_num = rand() % 10;

            if(i == 0 && rand_num == 0){
                rand_num = rand() % 10 + 1;
            }
            for (j = 0; j < i; j++){

                if (rand_num == rand_ans[j]){
                    break;
                }
            }
        } while (j < i);

        rand_ans[i] = rand_num;
        //printf("%d", rand_ans[i]); //デバック用
    }
}

int numCheck(const char user_ans[]){

    if(strlen(user_ans) != NUM){
        return 1;
    }

    for(int i = 0; i < NUM; i++){
        if(!isdigit(user_ans[i])){
            return 2;
        }
        for(int j = 0; j < i; j++){
            if(user_ans[i] == user_ans[j]){
                return 3;
            }
        }
    }
    return 0;
}

int readUserAns(const char user_ans[], const int com_ans[], int user_eat, int user_bite){
    user_eat = 0;
    user_bite = 0;

    for(int i = 0; i < NUM; i++){
        for(int j = 0; j < NUM; j++){
            if(user_ans[i] == '0' + com_ans[j]){
                if(i == j){
                    user_eat++;
                }else{
                    user_bite++;
                }
            }
        }
    }
    printf("eat: %d, bite: %d\n",user_eat , user_bite);
    return user_eat;
}

File: HitandBlow_Java/HitandBlow.java
import java.util.*;
import java.io.*;
import java.math.*;

public class HitandBlow {
    static Scanner sc = new Scanner(System.in);
    final static int answer_size = 3;
    static int eat = 0;
    static int bite = 0;
    static int cnt = 1;
    public static void main(String[] args) {
        
        System.out.println("*** Hit & Blow ***");
        int[] com_ans = randNumber();
        /*for(int a : com_ans){ //デバック用
            System.out.print(a);
        }*/
        int[] user_ans;

        while(true){
            user_ans = readUserAns();
            checkAns(com_ans,user_ans);
            cnt++;

            if(eat == answer_size){
                System.out.println("Congratulations!! ゲームを終了します");
                break;
            }
        }
    }

    public static int[] randNumber(){
        ArrayList<Integer> list = new ArrayList<>();
        int[] rand_ans = new int[answer_size];

        for(int i = 0; i < 10; i++){
            list.add(i);
        }

        while(true){
            Collections.shuffle(list);

            if(list.get(0) == 0){
                continue;
            }
            break;
        }

        for(int i = 0; i < rand_ans.length; i++){
            rand_ans[i] = list.get(i);
        }
        return rand_ans;
    }

    public static int[] readUserAns() {
        eat = 0;
        bite = 0;
        int[] ans = new int[answer_size];
        System.out.printf("%d回目のチャレンジ\n", cnt);
        String num;
        int user_num;

        for(int i = 0; i < ans.length; i++){

            try{
                System.out.printf("%dつ目の数字>>>",i+1);
                num = sc.nextLine();
                user_num = Integer.parseInt(num);
                ans[i] = user_num;
            } catch(NumberFormatException e){
                System.out.println("数字を入力してください");
                i--;
            }
        }
        return ans;
    }

    public static void checkAns(int[] com_ans, int[] user_ans){

        for(int i = 0; i < com_ans.length; i++){
            for(int j = 0; j < user_ans.length; j++){

                if(com_ans[j] == user_ans[i]){
                    if(j == i){
                        eat++;
                    }else{
                        bite++;
                    }
                }
            }
        }
        System.out.printf("eat: %d, bite: %d\n",eat,bite);
    } 
}
